julia/magic.py

```diff
@@ -66,6 +66,37 @@ def julia(self, line, cell=None):
         return ans
 
 
+def julia_completer(self, event):
+    pos = event.line.find("%julia")
+    if pos < 0:
+        return []
+    pos += len("%julia")  # pos: beginning of Julia code
+    julia_code = event.line[pos:]
+    julia_pos = len(event.text_until_cursor) - pos
+
+    julia = Julia()
+    completions = julia.eval("""
+    import REPL
+    (string, pos) -> begin
+        ret, _, should_complete =
+            REPL.completions(string, pos)
+        if should_complete
+            return map(REPL.completion_text, ret)
+        else
+            return String[]
+        end
+    end
+    """)(julia_code, julia_pos)
```

Comment:
should we `eval` this function once and cache it?

Reply:
Yes. I use `cached_property` in my `ipyjulia_hacks`. Something like this can be done here. https://github.com/tkf/ipyjulia_hacks/blob/679f6bbd3ed202cde6ec6ebb4b1b537566531cdf/src/ipyjulia_hacks/completers.py#L19

But could you review #190 first? I think that's more fundamental and important. But that PR, in turn, depends on #188.